compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -422,20 +422,41 @@ fn compute_hir_hash(
     })
 }
 
+/// Transforms Rust source code from an abstract syntax tree (AST) representation 
+/// into a higher-level intermediate representation (HIR).
+/// 
+/// ### Arguments:
+/// - TyCtxt (Type Context)
+/// - Empty tuple ()
+/// 
+/// ### Returns:
+/// - `hir::Crate` - represents the Higher-level Intermediate Representation 
+/// (HIR) of the entire crate being compiled.
 pub fn lower_to_hir(tcx: TyCtxt<'_>, (): ()) -> hir::Crate<'_> {
+    // extracting the compiler session
     let sess = tcx.sess;
     // Queries that borrow `resolver_for_lowering`.
+    // Precompute and store various pieces of information needed during the HIR lowering process
+    // - Determines the output filenames for the compilation
+    // - Performs early lint checks on the code.
+    // - Generates debugger visualizer information for the local crate.
     tcx.ensure_with_value().output_filenames(());
     tcx.ensure_with_value().early_lint_checks(());
     tcx.ensure_with_value().debugger_visualizers(LOCAL_CRATE);
+    // `resolver` variable helps resolve names and types during the compilation process
```

Comment:
I don't know if we need to explain what the resolver does. This should be the responsibility of the resolver documentation to explain what it does -- if that's lacking, then additional documentation should be put there, not here, since a function-internal comment doesn't have much discoverability.